apache/airflow
# Bugfix/26215 remove trigger UI glitches

This PR fixes some glitches that I found in the contribution of AIP-50 which slipped into the code at point of final review. I detected these during demo of the feature and hereby what to provide a fix.

Three glitches corrected with this PR:

- The DAG embedded javascript demo of the color slides in the example_params_ui_tutorial were broken
- The HTML dict formatting of the box/section "Generated Configuration JSON" in the new trigger UI (e.g. in "example_params_ui_tutorial" or other DAGs with a submit form was misaligned. setTimeout() event handler generates JavaScript error
- A TypeError was raised in JavaScript on first form submission when no "Select Recent Configurations" was rendered on page.

How to test that this fix works:

- Check out this branch, clean your dev env and re-compile www assets
- Start a local instance and open the browser to http://localhost:28080/trigger?dag_id=example_params_ui_tutorial
- See that (assuming you never ran the DAG) no JavaScript error is reported in JavaScript Developer Console
- Click on the gray section bar "Generated Configuration JSON" and see that dict formatting is "nice" (no mis-aligned text compared to main)
- Click on the gray section bar "Special advanced stuff with form fields" and use the sliders to adjust demo of color picker. See that color and text value adjusts when changing sliders and other way around if you change color code the sliders adjust. (was broken on main)

closes: #26215
